kernel/fdtable.c

```diff
@@ -374,6 +374,12 @@ int myst_fdtable_dup(
         new->device = old->device;
         new->object = newobj;
 
+        // add /proc/self/fd/[fd] entry for files
+        if (newfd != oldfd && new->type == MYST_FDTABLE_TYPE_FILE)
+        {
+            myst_add_fd_link(new->device, new->object, newfd);
```

Comment:
Can this operation fail? If it fails, do we need to clean up anything from fdtable?

Reply:
Have updated to closing the new file handle and forward the error.